<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1153</width>
    <height>635</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Заборов А.М. 224-322 Решение оптимизационных задач с помощью генетических алгоритмов</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>20</y>
      <width>111</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Размер популяции</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="population_count_spin_box">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>40</y>
      <width>71</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <number>3</number>
    </property>
    <property name="maximum">
     <number>500</number>
    </property>
    <property name="value">
     <number>100</number>
    </property>
   </widget>
   <widget class="QLabel" name="generations_count_spin_box_2">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>70</y>
      <width>141</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Количество поколений</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="generations_count_spin_box">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>90</y>
      <width>71</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
    <property name="maximum">
     <number>100</number>
    </property>
    <property name="value">
     <number>30</number>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>120</y>
      <width>141</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Вероятность мутации, %</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="mutation_probabilty_spin_box">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>140</y>
      <width>71</width>
      <height>22</height>
     </rect>
    </property>
    <property name="maximum">
     <number>100</number>
    </property>
    <property name="value">
     <number>10</number>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>150</x>
      <y>20</y>
      <width>120</width>
      <height>80</height>
     </rect>
    </property>
    <property name="title">
     <string>Режим работы</string>
    </property>
    <widget class="QRadioButton" name="step_mode_radio_button">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>82</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>Пошаговый</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QRadioButton" name="cycle_mode_radio_button">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>50</y>
       <width>101</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>Циклический</string>
     </property>
    </widget>
   </widget>
   <widget class="QSpinBox" name="graph_size_spin_box">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>200</y>
      <width>71</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <number>5</number>
    </property>
    <property name="maximum">
     <number>20</number>
    </property>
    <property name="value">
     <number>6</number>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>180</y>
      <width>151</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Количество вершин графа</string>
    </property>
   </widget>
   <widget class="QPushButton" name="find_shortest_path_button">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>410</y>
      <width>141</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Найти кратчайший путь</string>
    </property>
   </widget>
   <widget class="QPushButton" name="generate_graph_button">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>310</y>
      <width>141</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Сгенерировать граф</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_7">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>230</y>
      <width>151</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Значения рёбер</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="min_val_spin_box">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>250</y>
      <width>71</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
    <property name="value">
     <number>2</number>
    </property>
   </widget>
   <widget class="QLabel" name="label_8">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>250</y>
      <width>16</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>От</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_9">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>250</y>
      <width>16</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>до</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="max_val_spin_box">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>250</y>
      <width>71</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
    <property name="value">
     <number>20</number>
    </property>
   </widget>
   <widget class="QCheckBox" name="graph_fully_connected_check_box">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>280</y>
      <width>111</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Полносвязный</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>460</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Результат:</string>
    </property>
   </widget>
   <widget class="QLabel" name="result_label">
    <property name="geometry">
     <rect>
      <x>80</x>
      <y>460</y>
      <width>181</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QSpinBox" name="finish_node_spin_box">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>380</y>
      <width>71</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <number>0</number>
    </property>
    <property name="maximum">
     <number>19</number>
    </property>
    <property name="value">
     <number>5</number>
    </property>
   </widget>
   <widget class="QLabel" name="label_10">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>380</y>
      <width>16</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>От</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="start_node_spin_box">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>380</y>
      <width>71</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <number>0</number>
    </property>
    <property name="maximum">
     <number>19</number>
    </property>
    <property name="value">
     <number>0</number>
    </property>
   </widget>
   <widget class="QLabel" name="label_11">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>360</y>
      <width>151</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Путь</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_12">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>380</y>
      <width>16</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>до</string>
    </property>
   </widget>
   <widget class="QPushButton" name="prev_table_data_button">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>520</y>
      <width>111</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Пред. поколение</string>
    </property>
   </widget>
   <widget class="QPushButton" name="next_table_data_button">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>520</y>
      <width>111</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>След. поколение</string>
    </property>
   </widget>
   <widget class="QTableWidget" name="table">
    <property name="geometry">
     <rect>
      <x>280</x>
      <y>20</y>
      <width>851</width>
      <height>501</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>490</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Поколение</string>
    </property>
   </widget>
   <widget class="QLabel" name="current_gen_label">
    <property name="geometry">
     <rect>
      <x>80</x>
      <y>490</y>
      <width>41</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>1</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>830</x>
      <y>530</y>
      <width>301</width>
      <height>81</height>
     </rect>
    </property>
    <property name="text">
     <string>Заборов Артемий Михайлович
artem.zaborov@yandex.ru
https://online.mospolytech.ru/course/view.php?id=10055
02.04.2023</string>
    </property>
   </widget>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
